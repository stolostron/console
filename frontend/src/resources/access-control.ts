/* Copyright Contributors to the Open Cluster Management project */
import { IResource, IResourceDefinition } from './resource'

export const AccessControlApiVersion = 'rbac.open-cluster-management.io/v1alpha1'
export type AccessControlApiVersionType = 'rbac.open-cluster-management.io/v1alpha1'

export const AccessControlKind = 'ClusterPermission'
export type AccessControlKindType = 'ClusterPermission'

export const AccessControlDefinition: IResourceDefinition = {
  apiVersion: AccessControlApiVersion,
  kind: AccessControlKind,
}

export interface RoleBinding {
  namespace: string
  roleRef: RoleRef
  subject?: Subject
  subjects?: Subject[]
}

export interface ClusterRoleBinding {
  name?: string
  roleRef: RoleRef
  subject?: Subject
  subjects?: Subject[]
}

interface RoleRef {
  apiGroup: string
  kind: 'Role' | 'ClusterRole'
  name: string
}

interface Subject {
  apiGroup: string
  kind: 'User' | 'Group'
  name: string
  namespace?: string
}

export interface Condition {
  type: 'AppliedRBACManifestWork' | 'Validation'
  status: 'True' | 'False'
  reason?:
    | 'AppliedRBACManifestWork'
    | 'FailedValidationNoBindingsDefined'
    | 'FailedValidationNotInManagedClusterNamespace'
    | 'FailedBuildManifestWork'
  message: string
  lastTransitionTime: string
}

interface Status {
  conditions: Condition[]
}

export interface AccessControl extends IResource<Status> {
  apiVersion: AccessControlApiVersionType
  kind: AccessControlKindType
  spec: {
    roleBindings?: RoleBinding[]
    clusterRoleBinding?: ClusterRoleBinding
  }
}
