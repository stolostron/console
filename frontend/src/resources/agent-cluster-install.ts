/* Copyright Contributors to the Open Cluster Management project */
export const AgentClusterInstallKind = 'AgentClusterInstall'
export const AgentClusterInstallGroup = 'extensions.hive.openshift.io'
export const AgentClusterInstallVersion = `v1beta1`
export const AgentClusterInstallApiVersion = `${AgentClusterInstallGroup}/${AgentClusterInstallVersion}`
