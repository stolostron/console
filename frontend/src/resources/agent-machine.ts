/* Copyright Contributors to the Open Cluster Management project */
export const AgentMachineApiVersion = 'capi-provider.agent-install.openshift.io/v1alpha1'
export const AgentMachineKind = 'AgentMachine'
