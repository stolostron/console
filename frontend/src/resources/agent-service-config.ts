/* Copyright Contributors to the Open Cluster Management project */
export const AgentServiceConfigKind = 'AgentServiceConfig'
export const AgentServiceConfigKindVersion = 'agent-install.openshift.io/v1beta1'
