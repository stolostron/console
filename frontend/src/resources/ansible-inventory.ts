/* Copyright Contributors to the Open Cluster Management project */

export type AnsibleApiInventoryType = 'inventory'

export interface AnsibleTowerInventoryList {
  count?: number
  next?: string
  results: Array<AnsibleTowerInventory>
}
export interface AnsibleTowerInventory {
  name?: string
  type?: AnsibleApiInventoryType
}
