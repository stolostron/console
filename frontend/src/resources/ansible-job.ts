/* Copyright Contributors to the Open Cluster Management project */
import { Metadata } from './metadata'
import { IResourceDefinition } from './resource'
import { getLatest } from './utils/utils'

export const AnsibleJobApiVersion = 'tower.ansible.com/v1alpha1'
export type AnsibleJobApiVersionType = 'tower.ansible.com/v1alpha1'

export const AnsibleJobKind = 'AnsibleJob'
export type AnsibleJobKindType = 'AnsibleJob'

export type AnsibleJobTemplateType = 'Job' | 'Workflow'

export const AnsibleJobDefinition: IResourceDefinition = {
    apiVersion: AnsibleJobApiVersion,
    kind: AnsibleJobKind,
}

export interface AnsibleJob {
    apiVersion: AnsibleJobApiVersionType
    kind: AnsibleJobKindType
    metadata: Metadata
    spec?: {
        extra_vars: {}
    }
    status?: {
        ansibleJobResult: {
            changed: boolean
            failed: boolean
            status: string
            url: string
            finished: string
            started: string
        }
        conditions?: [
            {
                ansibleResult?: {
                    changed?: number
                    completion?: string
                    failures?: number
                    ok?: number
                    skipped?: number
                }
                lastTransitionTime?: string
                message?: string
                reason?: string
                status?: string
                type?: string
            }
        ]
    }
}

export function getLatestAnsibleJob(ansibleJobs: AnsibleJob[], namespace: string) {
    const jobs = ansibleJobs.filter((job) => job.metadata.namespace === namespace)

    const prehookJobs = getLatest<AnsibleJob>(
        jobs.filter((job) => job.metadata.annotations?.jobtype === 'prehook'),
        'status.ansibleJobResult.started'
    )
    const posthookJobs = getLatest<AnsibleJob>(
        jobs.filter((job) => job.metadata.annotations?.jobtype === 'posthook'),
        'status.ansibleJobResult.started'
    )

    return {
        prehook: prehookJobs,
        posthook: posthookJobs,
    }
}

export interface AnsibleTowerJobTemplateList {
    count?: number
    next?: string
    results?: Array<AnsibleTowerJobTemplate>
}
export interface AnsibleTowerJobTemplate {
    name?: string
    type?: AnsibleJobTemplateType
}
