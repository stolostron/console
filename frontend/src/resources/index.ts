/* Copyright Contributors to the Open Cluster Management project */
export * from './agent'
export * from './agent-cluster-install'
export * from './ansible-job'
export * from './application'
export * from './application-set'
export * from './argo-application'
export * from './bare-metal-asset'
export * from './bare-metal-host'
export * from './broker'
export * from './certificate-signing-requests'
export * from './channel'
export * from './cluster-claim'
export * from './cluster-curator'
export * from './cluster-deployment'
export * from './cluster-image-set'
export * from './cluster-management-add-on'
export * from './cluster-pool'
export * from './cluster-provision'
export * from './configmap'
export * from './cronjob'
export * from './daemonset'
export * from './deployment'
export * from './deployment-config'
export * from './discovered-cluster'
export * from './discovery-config'
export * from './gitops-cluster'
export * from './helmrelease'
export * from './infra-environment'
export * from './infrastructure'
export * from './install-config'
export * from './job'
export * from './klusterlet-add-on-config'
export * from './kustomization'
export * from './machine-pool'
export * from './managed-cluster'
export * from './managed-cluster-add-on'
export * from './managed-cluster-info'
export * from './managed-cluster-set'
export * from './managed-cluster-set-binding'
export * from './managedclusteraction'
export * from './managedclusterview'
export * from './multi-cluster-hub'
export * from './namespace'
export * from './nmstateconfig'
export * from './ocp-app-resource'
export * from './placement'
export * from './placement-binding'
export * from './placement-decision'
export * from './placement-rule'
export * from './pod'
export * from './policy'
export * from './policy-automation'
export * from './policy-report'
export * from './policy-set'
export * from './project'
export * from './provider-connection'
export * from './rbac'
export * from './resource'
export * from './secret'
export * from './selector'
export * from './self-subject-access-review'
export * from './stateful-set'
export * from './status'
export * from './submariner-config'
export * from './subscription'
export * from './subscription-operator'
export * from './subscription-report'
export * from './utils'
