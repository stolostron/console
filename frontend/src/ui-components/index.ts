/* Copyright Contributors to the Open Cluster Management project */

export * from './AcmActionGroup/AcmActionGroup'
export * from './AcmAlert'
export * from './AcmAutoRefreshSelect/AcmAutoRefreshSelect'
export * from './AcmBackdrop/AcmBackdrop'
export * from './AcmButton/AcmButton'
export * from './AcmCharts'
export * from './AcmCountCard'
export * from './AcmCodeSnippet/AcmCodeSnippet'
export * from './AcmDescriptionList/AcmDescriptionList'
export * from './AcmDrawer/AcmDrawer'
export * from './AcmDropdown/AcmDropdown'
export * from './AcmLaunchLink/AcmLaunchLink'
export * from './AcmModal/AcmModal'
export * from './AcmEmptyState/AcmEmptyState'
export * from './AcmExpandable'
export * from './AcmErrorBoundary/AcmErrorBoundary'
export * from './AcmIcons/AcmIcons'
export * from './AcmInlineCopy/AcmInlineCopy'
export * from './AcmInlineStatus'
export * from './AcmForm/AcmForm'
export * from './AcmLabels/AcmLabels'
export * from './AcmLabelsInput/AcmLabelsInput'
export * from './AcmLoadingPage/AcmLoadingPage'
export * from './AcmLogWindow/AcmLogWindow'
export * from './AcmModal/AcmModal'
export * from './AcmNumberInput/AcmNumberInput'
export * from './AcmPage/AcmPage'
export * from './AcmPageProcess/AcmPageProcess'
export * from './AcmProvider'
export * from './AcmRefreshTime/AcmRefreshTime'
export * from './AcmSearchbar/AcmSearchbar'
export * from './AcmSecondaryNav/AcmSecondaryNav'
export * from './AcmSelect/AcmSelect'
export * from './AcmSummaryList/AcmSummaryList'
export * from './AcmTable/AcmTable'
export * from './AcmTextInput/AcmTextInput'
export * from './AcmTextArea/AcmTextArea'
export * from './AcmTile/AcmTile'
export * from './AcmFormSection/AcmFormSection'
export * from './AcmScrollable/AcmScrollable'
export * from './AcmHeader/AcmHeader'
export * from './AcmMultiSelect/AcmMultiSelect'
export * from './AcmProgressTracker/AcmProgressTracker'
