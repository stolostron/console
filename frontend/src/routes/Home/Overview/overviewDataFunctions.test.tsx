/* Copyright Contributors to the Open Cluster Management project */
import i18next from 'i18next'
import { Cluster, ClusterStatus } from '../../../resources'
import { Provider } from '../../../ui-components'
import { mockRequestedCounts } from '../../Applications/Application.sharedmocks'
import {
  getAddonHealth,
  getAppTypeSummary,
  getClusterProviderSummary,
  getClusterStatus,
  getClusterVersionSummary,
  getComplianceData,
  getFilteredClusters,
  getNodeSummary,
  getPolicyReport,
  getPolicySummary,
  getWorkerCoreTotal,
  parseAlertsMetric,
  parseOperatorMetric,
  parseUpgradeRiskPredictions,
} from './overviewDataFunctions'
import {
  mockAlertMetrics,
  mockOperatorMetrics,
  mockWorkerCoreCountMetrics,
  parsedAddons,
  policies,
  policyReports,
} from './sharedmocks'

const filteredClusterNames: string[] = ['managed', 'local-cluster']
const clusterData: Cluster[] = [
  {
    name: 'managed-1',
    displayName: 'managed-1',
    namespace: 'managed-1',
    uid: 'd75e89bc-29d7-45ca-b057-bb84dc095200',
    status: ClusterStatus.unknown,
    statusMessage: 'Registration agent stopped updating its lease.',
    provider: Provider.aws,
    labels: {
      cloud: 'Amazon',
      'cluster.open-cluster-management.io/clusterset': 'default',
      env: 'dev',
      name: 'managed-1',
      openshiftVersion: '4.13.4',
      'openshiftVersion-major': '4',
      'openshiftVersion-major-minor': '4.13',
      vendor: 'OpenShift',
    },
    nodes: { nodeList: [], ready: 0, unhealthy: 0, unknown: 0 },
    isHive: false,
    isHypershift: false,
    isManaged: true,
    isCurator: false,
    hasAutomationTemplate: false,
    isHostedCluster: false,
    isSNOCluster: false,
    isRegionalHubCluster: false,
    hive: { isHibernatable: false, secrets: {} },
    clusterSet: 'default',
    owner: {},
  },
  {
    name: 'local-cluster',
    displayName: 'local-cluster',
    namespace: 'local-cluster',
    uid: 'a23a4032-0ba7-42fd-9439-f7b7667c9f77',
    status: ClusterStatus.ready,
    provider: Provider.aws,
    distribution: {
      isManagedOpenShift: false,
      displayVersion: 'OpenShift 4.15.19',
    },
    labels: {
      cloud: 'Amazon',
      'cluster.open-cluster-management.io/clusterset': 'default',
      'local-cluster': 'true',
      name: 'local-cluster',
      openshiftVersion: '4.13.4',
      'openshiftVersion-major': '4',
      'openshiftVersion-major-minor': '4.13',
      'velero.io/exclude-from-backup': 'true',
      vendor: 'OpenShift',
      clusterID: 'local-cluster',
    },
    nodes: {
      nodeList: [
        {
          capacity: { cpu: '16', memory: '64453824Ki' },
          conditions: [{ status: 'True', type: 'Ready' }],
          labels: {
            'beta.kubernetes.io/instance-type': 'm6a.4xlarge',
            'failure-domain.beta.kubernetes.io/region': 'us-east-1',
            'failure-domain.beta.kubernetes.io/zone': 'us-east-1a',
            'node-role.kubernetes.io/control-plane': '',
            'node-role.kubernetes.io/master': '',
            'node-role.kubernetes.io/worker': '',
            'node.kubernetes.io/instance-type': 'm6a.4xlarge',
          },
          name: 'ip-10-0-140-19.ec2.internal',
        },
      ],
      ready: 1,
      unhealthy: 0,
      unknown: 0,
    },
    kubeApiServer: 'https://api.sno-4xlarge-413-j4tls.dev07.red-chesterfield.com:6443',
    consoleURL: 'https://console-openshift-console.apps.sno-4xlarge-413-j4tls.dev07.red-chesterfield.com',
    isHive: false,
    isHypershift: false,
    isManaged: true,
    isCurator: false,
    hasAutomationTemplate: false,
    isHostedCluster: false,
    isSNOCluster: false,
    isRegionalHubCluster: false,
    hive: { isHibernatable: false, secrets: {} },
    clusterSet: 'default',
    owner: {},
    creationTimestamp: '2023-07-24T13:00:53Z',
  },
]

test('Correctly returns getFilteredClusters', () => {
  const result = getFilteredClusters(clusterData, { env: ['dev'] })
  expect(result).toMatchSnapshot()
})

test('Correctly returns getClusterProviderSummary', () => {
  const result = getClusterProviderSummary(clusterData)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getClusterVersionSummary', () => {
  const result = getClusterVersionSummary(clusterData)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getAppTypeSummary', () => {
  const t = i18next.t.bind(i18next)
  const result = getAppTypeSummary(mockRequestedCounts, t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getPolicySummary', () => {
  const t = i18next.t.bind(i18next)
  const result = getPolicySummary(policies, filteredClusterNames, 2, t)
  expect(result).toMatchSnapshot()
})
test('Correctly returns getPolicySummary with filtered clusters', () => {
  const t = i18next.t.bind(i18next)
  const result = getPolicySummary(policies, [filteredClusterNames[1]], 2, t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getNodeSummary', () => {
  const t = i18next.t.bind(i18next)
  const result = getNodeSummary(clusterData, t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getWorkerCoreTotal', () => {
  const result = getWorkerCoreTotal(mockWorkerCoreCountMetrics, clusterData)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getPolicyReport', () => {
  const result = getPolicyReport(policyReports, clusterData)
  expect(result).toMatchSnapshot()
})

test('Correctly returns parseUpgradeRiskPredictions with no predictions', () => {
  const result = parseUpgradeRiskPredictions([])
  expect(result).toMatchSnapshot()
})

test('Correctly returns parseUpgradeRiskPredictions with predictions', () => {
  const result = parseUpgradeRiskPredictions([
    {
      cluster_id: 'test-cluster-id-1',
      prediction_status: 'ok',
      upgrade_recommended: true,
      upgrade_risks_predictors: {
        alerts: [],
        operator_conditions: [],
      },
      last_checked_at: '2024-03-27T14:00:19.137279+00:00',
    },
    {
      cluster_id: 'test-cluster-id-2',
      upgrade_recommended: false,
      upgrade_risks_predictors: {
        alerts: [
          {
            name: 'ClusterOperatorDown',
            namespace: 'openshift-cluster-version',
            severity: 'critical',
            url: 'https://console-openshift-console.com/testing',
          },
          {
            name: 'NodeClockNotSynchronising',
            namespace: 'openshift-monitoring',
            severity: 'warning',
            url: 'https://console-openshift-console.com/testing',
          },
        ],
        operator_conditions: [],
      },
      last_checked_at: '2024-03-27T14:00:19.137279+00:00',
    },
    {
      cluster_id: 'test-cluster-id-3',
      prediction_status: 'No data for the cluster',
    },
    {
      cluster_id: 'test-cluster-id-4',
      upgrade_recommended: false,
      upgrade_risks_predictors: {
        alerts: [
          {
            name: 'ClusterOperatorDown',
            namespace: 'openshift-cluster-version',
            severity: 'info',
            url: 'https://console-openshift-console.com/testing',
          },
        ],
        operator_conditions: [],
      },
      last_checked_at: '2024-03-27T14:00:19.137279+00:00',
    },
  ])
  expect(result).toMatchSnapshot()
})

test('Correctly returns getClusterStatus', () => {
  const t = i18next.t.bind(i18next)
  const result = getClusterStatus(clusterData, 'label%3Acloud%3DAmazon', t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getComplianceData', () => {
  const t = i18next.t.bind(i18next)
  const result = getComplianceData(clusterData, filteredClusterNames, policies, t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns getAddonHealth', () => {
  const t = i18next.t.bind(i18next)
  const result = getAddonHealth(parsedAddons, filteredClusterNames, t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns parseAlertsMetric', () => {
  const t = i18next.t.bind(i18next)
  const result = parseAlertsMetric(mockAlertMetrics, [], t)
  expect(result).toMatchSnapshot()
})

test('Correctly returns parseOperatorMetric', () => {
  const result = parseOperatorMetric(mockOperatorMetrics, [])
  expect(result).toMatchSnapshot()
})
