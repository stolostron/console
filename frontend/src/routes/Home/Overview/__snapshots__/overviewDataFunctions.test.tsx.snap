// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly returns getAddonHealth 1`] = `
Array [
  Object {
    "key": "Degraded",
    "link": "/multicloud/infrastructure/clusters/managed?addons=Degraded",
    "value": 1,
  },
  Object {
    "key": "Progressing",
    "link": "/multicloud/infrastructure/clusters/managed?addons=Progressing",
    "value": 1,
  },
  Object {
    "key": "Unknown",
    "link": "/multicloud/infrastructure/clusters/managed?addons=Unknown",
    "value": 3,
  },
  Object {
    "isPrimary": true,
    "key": "Available",
    "link": "/multicloud/infrastructure/clusters/managed?addons=Available",
    "value": 3,
  },
]
`;

exports[`Correctly returns getClusterStatus 1`] = `
Array [
  Object {
    "key": "Offline",
    "link": "/multicloud/search?filters={\\"textsearch\\":\\"kind%3ACluster%20ManagedClusterConditionAvailable%3A!True%20label%3Acloud%3DAmazon\\"}",
    "value": 1,
  },
  Object {
    "isPrimary": true,
    "key": "Ready",
    "link": "/multicloud/search?filters={\\"textsearch\\":\\"kind%3ACluster%20ManagedClusterConditionAvailable%3ATrue%20label%3Acloud%3DAmazon\\"}",
    "value": 1,
  },
]
`;

exports[`Correctly returns getClustersSummary 1`] = `
Array [
  Object {
    "count": 2,
    "icon": undefined,
    "id": "total-clusters",
    "link": "/multicloud/infrastructure/clusters/managed",
    "title": "Clusters",
  },
  Object {
    "count": 4,
    "icon": undefined,
    "id": "apps-count",
    "link": "/multicloud/applications",
    "title": "Applications",
  },
  Object {
    "count": 1,
    "icon": <DatabaseIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />,
    "id": "kube-types",
    "link": undefined,
    "title": "Kubernetes type",
  },
  Object {
    "count": 1,
    "icon": <MapPinIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />,
    "id": "cluster-regions",
    "link": undefined,
    "title": "Region",
  },
  Object {
    "count": 1,
    "icon": <CubesIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />,
    "id": "nodes-count",
    "link": "/multicloud/search?filters={\\"textsearch\\":\\"kind%3ANode\\"}",
    "title": "Nodes",
  },
]
`;

exports[`Correctly returns getComplianceData 1`] = `
Array [
  Object {
    "key": "Violations",
    "link": "/multicloud/governance/policies?violations=violations",
    "value": 1,
  },
  Object {
    "isPrimary": true,
    "key": "No violations",
    "link": "/multicloud/governance/policies?violations=no-violations&violations=no-status",
    "value": 1,
  },
]
`;

exports[`Correctly returns getFilteredClusters 1`] = `
Array [
  Object {
    "clusterSet": "default",
    "displayName": "managed-1",
    "hasAutomationTemplate": false,
    "hive": Object {
      "isHibernatable": false,
      "secrets": Object {},
    },
    "isCurator": false,
    "isHive": false,
    "isHostedCluster": false,
    "isHypershift": false,
    "isManaged": true,
    "isRegionalHubCluster": false,
    "isSNOCluster": false,
    "labels": Object {
      "cloud": "Amazon",
      "cluster.open-cluster-management.io/clusterset": "default",
      "env": "dev",
      "name": "managed-1",
      "openshiftVersion": "4.13.4",
      "openshiftVersion-major": "4",
      "openshiftVersion-major-minor": "4.13",
      "vendor": "OpenShift",
    },
    "name": "managed-1",
    "namespace": "managed-1",
    "nodes": Object {
      "nodeList": Array [],
      "ready": 0,
      "unhealthy": 0,
      "unknown": 0,
    },
    "owner": Object {},
    "provider": "aws",
    "status": "unknown",
    "statusMessage": "Registration agent stopped updating its lease.",
    "uid": "d75e89bc-29d7-45ca-b057-bb84dc095200",
  },
]
`;

exports[`Correctly returns getNodeCount 1`] = `1`;

exports[`Correctly returns getPolicyReport 1`] = `
Object {
  "clustersWithIssuesCount": 1,
  "policyReportCriticalCount": 1,
  "policyReportImportantCount": 1,
  "policyReportLowCount": 1,
  "policyReportModerateCount": 1,
}
`;

exports[`Correctly returns parseUpgradeRiskPredictions with no predictions 1`] = `
Object {
  "clustersWithRiskPredictors": 0,
  "criticalUpdateCount": 0,
  "infoUpdateCount": 0,
  "warningUpdateCount": 0,
}
`;

exports[`Correctly returns parseUpgradeRiskPredictions with predictions 1`] = `
Object {
  "clustersWithRiskPredictors": 2,
  "criticalUpdateCount": 1,
  "infoUpdateCount": 1,
  "warningUpdateCount": 1,
}
`;

exports[`Correctly returns parseAlertsMetric 1`] = `
Object {
  "alertSeverity": Object {
    "critical": Object {
      "alerts": Array [],
      "icon": <CriticalRiskIcon />,
      "key": "critical",
      "label": "Critical",
    },
    "info": Object {
      "alerts": Array [
        "InsightsRecommendationActive",
        "InsightsRecommendationActive",
      ],
      "icon": undefined,
      "key": "info",
      "label": "Info",
    },
    "other": Object {
      "alerts": Array [],
      "icon": undefined,
      "key": "other",
      "label": "Other",
    },
    "warning": Object {
      "alerts": Array [
        "AlertmanagerReceiversNotConfigured",
      ],
      "icon": <ImportantRiskIcon />,
      "key": "warning",
      "label": "Warning",
    },
  },
  "clustersAffectedAlerts": Array [
    "local-cluster",
    "managed-1",
  ],
}
`;

exports[`Correctly returns parseOperatorMetric 1`] = `
Object {
  "clustersAffectedOperator": Array [],
  "degraded": Array [
    "etcd",
  ],
  "notAvailable": Array [
    "baremetal",
  ],
  "other": Array [
    "cloud-controller-manager",
    "cloud-credential",
    "cloud-credential",
  ],
}
`;
